packages/apollo-server-defer/README.md

```diff
@@ -0,0 +1,173 @@
+    "position": 20
+  }
+}
+{
+  "path": ["video", "comments"],
+  "data": [{
+    "userId": "123",
+    "text": "..."
+  }]
+}
+```
+
+#### Who needs this the most?
+
+- Media companies, banks, other data-heavy enterprise applications.
+
+#### Why is @defer an ideal solution?
+
+- Query remains super easy to read 👍
+- Stays true to Apollo's declarative approach to data loading - everything is encapsulated in a single query component 😎
+- Greatly reduces boilerplate code to manage multiple requests, a huge value-add for users that use both Apollo Server AND Client together.
+
+### Implementation Details
+
+#### Patch Data Format:
+
+- Proposed format for `ExecutionPatchResult`
+  ```graphql
+  {
+    path: [string]!
+    data: {}
+    errors: []
+  }
+  ```
+- Current transport assumes ExecutionResult so some changes need to be made.
+- We should use the same patch format for @defer/stream/live support.
+
+#### Apollo Server:
+
+In order to support @defer, there are significant changes to be made to the execution phase if GraphQL.
+
+- Roll our own derivative of graphql.js execution that supports deferred responses and streaming patches using observables.
+- Try to maximize code reuse by exporting types and utility functions from graphql.js, making a PR if necessary.
+- Restrict peer dependency on graphql.js to versions that we have tested with.
+- @skip and @include should take priority over @defer
+
+#### Apollo Client:
+
+- No changes required
+- Initial implementation of @defer support should come as a Apollo Link. Reads from a socket connection or some other event stream, keeps the partial response in memory, merging patches as they come and pushing it through the link stack.
+
+#### Errors:
+
+This refers specifically to the errors that occur when streaming deferred fields. Errors on the initial response will be handled as per normal.
+
+- Resolver level errors can be sent along with the patch. These will be merged with the errors array to create the full response.
+- Network errors (i.e when the websocket connection is broken). Ideally, a network error should not nullify the results that have already been sent succesfully. However, the client should expose the network status to indicate whether all the patches have been sent successfully.  
+
+#### Transport:
+
+- @defer requires a transport layer that is able to stream or push data to the client, like websockets or server side events. If the transport layer does not support this, Apollo Server should fallback to normal execution and ignore @defer.
+- Note: Deferred queries are not able to be refetched in isolation. So links like apollo-link-retry and apollo-link-error might need to ignore patches for now.
+
+#### Restrictions on @defer usage:
+
+- _Mutations:_ already execute serially, so there does not seem to be any use case for @defer. We should throw an error if @defer is applied to the root mutation type. However, @defer should apply normally for the selection set on a mutation.
```

Comment:
> However, @defer should apply normally for the selection set on a mutation.

I think we maybe decided that defer shouldn't be allowed anywhere in a mutation?

Also, I think that should also apply to subscriptions, at least at first to simplify things.

Reply:
Ah yes! Because there is no analog to watchQuery on the client! Can do!